core/src/main/java/com/zextras/carbonio/files/graphql/datafetchers/NodeDataFetcher.java

```diff
@@ -886,6 +886,25 @@ public DataFetcher<CompletableFuture<List<DataFetcherResult<Map<String, Object>>
                   }
                 });
             }
+
+            // check if node needs an alternative name
+            List<String> targetFolderChildrenFilesName = nodeRepository.getNodes(
+                  nodeRepository.getChildrenIds(
```

Comment:
I would replace this two database calls with a single one using the `nodeRepository.findNodes()` method